### PR TITLE
DOC: clarify min supported version wording

### PR DESCRIPTION
**Pull request by tacaswell**
- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/devel/min_dep_policy.rst [hunk 1]
```diff
@@ -4,30 +4,36 @@
...
 
-- support minor versions of ``Python`` initially released in the previous
-  3 years
-- support minor versions of ``numpy`` initially released in the
-  previous 3 years or oldest that supports the minimum python version
-  (which ever is higher)
+- support minor versions of ``Python`` initially released
+  36 months prior to our planned release date.
+- support minor versions of ``numpy`` initially released in the 36
+  months prior to our planned release dat or oldest that supports the
```
Comment (anntzer) [inline doc/devel/min_dep_policy.rst]: date (typo)

File: doc/devel/min_dep_policy.rst [hunk 2]
```diff
@@ -4,30 +4,36 @@
...
 
 Python Dependencies
 ===================
 
-For python dependencies we should support at least
+For python dependencies we should support at least:
 
 with compiled extensions
-  minor versions released in the last 3 years
+  minor versions initially released in the 36 months prior to our release
```
Comment (anntzer) [inline doc/devel/min_dep_policy.rst]: planned release

File: doc/devel/min_dep_policy.rst [hunk 3]
```diff
@@ -4,30 +4,36 @@
...
   or the oldest that support our minimum python + numpy
 
 without complied extensions
-  minor versions released in the last 2 years or the oldest that
-  supports our minimum python.
+  minor versions initially released in the 24 months prior to our release
```
Comment (anntzer) [inline doc/devel/min_dep_policy.rst]: planned release

**Comment by tacaswell** [comment]
force-pushed to fix typos / omissions.

File: doc/devel/min_dep_policy.rst [hunk 4]
```diff
@@ -4,34 +4,56 @@
...
+
 Python and numpy
 ================
 
-- support minor versions of ``Python`` initially released in the previous
-  3 years
-- support minor versions of ``numpy`` initially released in the
-  previous 3 years or oldest that supports the minimum python version
-  (which ever is higher)
+- support minor versions of ``Python`` initially released
```
Comment (QuLogic) [inline doc/devel/min_dep_policy.rst]: This should have 'in the' like the NumPy line, no? Right now it reads as if only Python older than 3 years is supported and nothing newer.

**Comment by NelleV** [comment]
LGTM :+1: 
It's a clear improvement on our current documentation, so I'm going to go ahead and merge this. We can always improve the wording afterwards.

**Pull request merged**